UX is poor when ssh-agent misconfigured

`fleetctl` is not very helpful when your ssh-agent is misconfigured:

```
$ fleetctl status apache.1.service
2014/02/12 19:07:48 Unable to execute command over SSH: dial unix: missing address
```

```
$ fleetctl-imc list-machines
panic: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain

goroutine 1 [running]:
runtime.panic(0x2b1920, 0x2108979e0)
    /usr/local/go/src/pkg/runtime/panic.c:266 +0xb6
main.getRegistry(0x21086eee0, 0x2108710d0)
    /home/philips/coreos/fleet/fleet/src/github.com/coreos/fleet/fleetctl/cmd.go:51 +0x1ab
main.listMachinesAction(0x21086eee0)
    /home/philips/coreos/fleet/fleet/src/github.com/coreos/fleet/fleetctl/list_machines.go:31 +0x35
github.com/coreos/fleet/third_party/github.com/codegangsta/cli.Command.Run(0x347eb0, 0xd, 0x0, 0x0, 0x38fe10, ...)
    /home/philips/coreos/fleet/fleet/src/github.com/coreos/fleet/third_party/github.com/codegangsta/cli/command.go:73 +0x994
github.com/coreos/fleet/third_party/github.com/codegangsta/cli.(*App).Run(0x21088e000, 0x21082d000, 0x4, 0x4, 0x7, ...)
    /home/philips/coreos/fleet/fleet/src/github.com/coreos/fleet/third_party/github.com/codegangsta/cli/app.go:111 +0x855
main.main()
    /home/philips/coreos/fleet/fleet/src/github.com/coreos/fleet/fleetctl/cmd.go:99 +0x9c4

goroutine 3 [chan receive]:
github.com/coreos/fleet/third_party/github.com/golang/glog.(*loggingT).flushDaemon(0x6dad80)
    /home/philips/coreos/fleet/fleet/src/github.com/coreos/fleet/third_party/github.com/golang/glog/glog.go:839 +0x50
created by github.com/coreos/fleet/third_party/github.com/golang/glog.init·1
    /home/philips/coreos/fleet/fleet/src/github.com/coreos/fleet/third_party/github.com/golang/glog/glog.go:406 +0x276
Alexs-MacBook-Air-3:Downloads polvi$ ssh-add ~/.ssh/polvi-air2.pem 
Identity added: /Users/polvi/.ssh/polvi-air2.pem (/Users/polvi/.ssh/polvi-air2.pem)
Alexs-MacBook-Air-3:Downloads polvi$ fleetctl-imc list-machines
MACHINE     IP      METADATA
...
```

Here are some methods of being more helpful:

```
$ fleetctl status apache.1.service
Unable to make an SSH connection because there is no ssh-agent running. Ensure than an agent
is running on this machine or forward your ssh agent from your local machine before
connecting to the remote machine. Example: "ssh -A core@1.2.3.4"
```

```
$ fleetctl list-machines
ssh: unable to authenticate via ssh agent. 
Maybe ssh-add ~/.ssh/my-key.pem and try again?
```

--- a/fleetctl/cmd.go
+++ b/fleetctl/cmd.go
@@ -61,7 +61,7 @@ func getRegistry(context *cli.Context) *registry.Registry {
 		}
 
 		tr := http.Transport{
-			Dial:	dial,
+			Dial: dial,
 			TLSClientConfig: &tls.Config{
 				InsecureSkipVerify: true,
 			},

--- a/ssh/ssh.go
+++ b/ssh/ssh.go
@@ -2,6 +2,7 @@ package ssh
 
 import (
 	"bufio"
+	"errors"
 	"net"
 	"os"
 
@@ -62,29 +63,41 @@ func Shell(client *gossh.ClientConn) error {
 	return nil
 }
 
-func sshClientConfig(user string) *gossh.ClientConfig {
-	agent, err := net.Dial("unix", os.Getenv("SSH_AUTH_SOCK"))
+func sshClientConfig(user string) (*gossh.ClientConfig, error) {
+	sock := os.Getenv("SSH_AUTH_SOCK")
+	if sock == "" {
+		return nil, errors.New("SSH_AUTH_SOCK environment variable is not set. Verify ssh-agent is running. See https://github.com/coreos/fleet/blob/master/Documentation/remote-access.md for help.")
+	}
+
+	agent, err := net.Dial("unix", sock)
 	if err != nil {
-		return nil
+		return nil, err
 	}
 
-	auths := []gossh.ClientAuth{
-		gossh.ClientAuthAgent(gossh.NewAgentClient(agent)),
-	}
-
-	return &gossh.ClientConfig{
+	cfg := gossh.ClientConfig{
 		User:	user,
-		Auth:	auths,
+		Auth:	[]gossh.ClientAuth{
+			gossh.ClientAuthAgent(gossh.NewAgentClient(agent)),
+		},
 	}
+
+	return &cfg, nil
 }
 
 func NewSSHClient(user, addr string) (*gossh.ClientConn, error) {
-	clientConfig := sshClientConfig(user)
+	clientConfig, err := sshClientConfig(user)
+	if err != nil {
+		return nil, err
+	}
+
 	return gossh.Dial("tcp", addr, clientConfig)
 }
 
 func NewTunnelledSSHClient(user, tunaddr, tgtaddr string) (*gossh.ClientConn, error) {
-	clientConfig := sshClientConfig(user)
+	clientConfig, err := sshClientConfig(user)
+	if err != nil {
+		return nil, err
+	}
 
 	tunnelClient, err := gossh.Dial("tcp", tunaddr, clientConfig)
 	if err != nil {

